src/main/java/bio/overture/riff/service/RiffService.java

```diff
@@ -75,7 +75,7 @@ public RiffResponse makeRiff(JWTUser user, ShortenRequest request) {
   }
 
   public boolean deleteRiff(JWTUser user, String id) {
-    val optionalRiff = repository.findById(Long.valueOf(id, 36));
+    val optionalRiff = repository.findByUidAndId(user.getUid(), Long.valueOf(id, 36));
     if (optionalRiff.isPresent()) {
       val riff = optionalRiff.get();
       if (user.getUid().equals(riff.getUid())) {
```

Comment:
Looks like this line provides the logic to confirm that the riff entry matches the user id from the request. If so, we don't need any modification tot he repository fetch request.

Perhaps all we needed to do was add a check like this to the updateRiff method.

Reply:
I missed that, I'll change the code to have the same thing than the delete